src/connector/src/source/datagen/source/generator.rs

```diff
@@ -197,20 +210,13 @@ impl DatagenEventGenerator {
                         row.push(datum);
                     }
 
-                    rows.push((Op::Insert, OwnedRow::new(row)));
                     self.offset += 1;
+                    rows.push((Op::Insert, OwnedRow::new(row)));
                     rows_generated_this_second += 1;
                 }
 
                 if !rows.is_empty() {
-                    let chunk = StreamChunk::from_rows(&rows, &self.data_types);
-                    let mapping = hashmap! {
-                        self.split_id.clone() => (self.offset - 1).to_string()
```

Comment:
Why did we `-1` here previously?

Reply:
I think it was a wrong implementation. Offset should be updated before yielded, otherwise cases of different rows with same offset may happen.